packages/medical-domain/src/domains/repositories/statistics/glycemia-statistics.service.ts

```diff
@@ -0,0 +1,278 @@
+  const totalDuration = getCgmTotalDuration(filteredCbg)
+
+  const total = Math.round(numDays * MS_IN_DAY)
+
+  return {
+    sensorUsage: totalDuration,
+    total
+  }
+}
+
+function getAverageGlucoseData(bgData: Cbg[] | Smbg[], dateFilter: DateFilter): AverageGlucoseStatistics {
+  const filteredBg = CbgService.filterOnDate(bgData as Cbg[], dateFilter.start, dateFilter.end, getWeekDaysFilter(dateFilter)) as Bg[]
+  const totalBgValue = filteredBg.reduce((sum, bgData) => sum + bgData.value, 0)
+  const numberOfBg = filteredBg.length
+  return {
+    averageGlucose: totalBgValue / numberOfBg,
+    total: numberOfBg
+  }
+}
+
+function getCoefficientOfVariationData(bgData: Cbg[] | Smbg[], dateFilter: DateFilter): CoefficientOfVariationStatistics {
+  const filteredBg = CbgService.filterOnDate(bgData as Cbg[], dateFilter.start, dateFilter.end, getWeekDaysFilter(dateFilter))
+  const average = getAverageGlucoseData(filteredBg, dateFilter)
+  const noDataStatistics = {
+    insufficientData: true,
+    coefficientOfVariation: Number.NaN,
+    total: average.total
+  }
+  if (average.total < 3) {
+    return noDataStatistics
+  }
+
+  const initialAcc: Record<string, number[]> = {}
+  const bgValueByDay = filteredBg.reduce((current, bgData) => {
+    if (current[bgData.localDate] === undefined) {
+      current[bgData.localDate] = [bgData.value]
+      return { ...current }
+    }
+    current[bgData.localDate].push(bgData.value)
+    return { ...current }
+  }, initialAcc)
+
+  const coefficientOfVariationByDate = Object.keys(bgValueByDay).reduce((current: number[], localDate): number[] => {
+    const numBg = bgValueByDay[localDate].length
+    if (numBg >= 3) {
+      const avgGlucose = meanValues(bgValueByDay[localDate])
+      const squaredDiffs = bgValueByDay[localDate].map((bgValue: number) => (bgValue - avgGlucose) ** 2)
+      const sumSquaredDiffs = sumValues(squaredDiffs)
+      const standardDeviation = Math.sqrt(sumSquaredDiffs / (numBg - 1))
+      return [...current, standardDeviation / avgGlucose * 100]
+    }
+    return [...current]
+  }, [])
+  if (coefficientOfVariationByDate.length === 0) {
+    return noDataStatistics
+  }
+  return {
+    insufficientData: false,
+    coefficientOfVariation: meanValues(coefficientOfVariationByDate),
+    total: average.total
+  }
+}
+
+function getGlucoseManagementIndicatorData(cbgData: Cbg[], bgUnit: BgUnit, dateFilter: DateFilter): GlucoseManagementIndicatoStatistics {
+  const insufficientData = {
+    glucoseManagementIndicator: Number.NaN,
+    total: Number.NaN,
+    insufficientData: true
+  }
+  if (diffDays(dateFilter.start, dateFilter.end) < 14) {
+    return insufficientData
+  }
+
+  const filteredCbg = CbgService.filterOnDate(cbgData, dateFilter.start, dateFilter.end, getWeekDaysFilter(dateFilter))
+  const totalCbgDuration = getCgmTotalDuration(filteredCbg)
+  if (totalCbgDuration < 14 * MS_IN_DAY * 0.7) {
```

Comment:
What is `0.7`?

Reply:
70%  
i.e. duration must be at least 70% of 14 days. You'd rather have a const seventyPercent ?

Reply:
Yeah, SEVENTY_PERCENT would make me feel happy